internal/mining/mining.go

```diff
@@ -40,6 +40,134 @@ const (
 	MinHighPriority = dcrutil.AtomsPerCoin * 144.0 / 250
 )
 
+// Config is a descriptor containing the mining configuration.
+type Config struct {
+	// Policy houses the policy (configuration parameters) which is used to control
+	// the generation of block templates.
+	Policy *Policy
+
+	// TxSource represents a source of transactions to consider for inclusion in
+	// new blocks.
+	TxSource TxSource
+
+	// TimeSource defines a time source to use.
+	TimeSource blockchain.MedianTimeSource
+
+	// SubsidyCache defines a subsidy cache to use.
+	SubsidyCache *standalone.SubsidyCache
+
+	// ChainParams defines the chain parameters to use.
+	ChainParams *chaincfg.Params
+
+	// BlockManager defines the block manager to use.
+	BlockManager blockManagerFacade
```

Comment:
Considering the detail provided for the other config fields these could be more descriptive.